fwoa-tools/src/binaryConverter.ts

```diff
@@ -0,0 +1,145 @@
+import { readFileSync } from 'fs';
+import { S3 } from 'aws-sdk';
+import { GetObjectOutput } from 'aws-sdk/clients/s3';
+import * as dotenv from 'dotenv';
+import yargs from 'yargs';
+import { ExportOutput } from './migrationUtils';
+
+dotenv.config({ path: '.env' });
+const { EXPORT_BUCKET_NAME, BINARY_BUCKET_NAME, API_AWS_REGION } = process.env;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function parseCmdOptions(): any {
+  return yargs(process.argv.slice(2))
+    .usage('Usage: $0 [--dryRun, -d boolean]')
+    .describe('dryRun', 'Check operations and authentication status')
+    .boolean('dryRun')
+    .default('dryRun', false)
+    .alias('d', 'dryRun').argv;
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const argv: any = parseCmdOptions();
+const dryRun: boolean = argv.dryRun;
+const exportBucketName: string | undefined = process.env.EXPORT_BUCKET_NAME;
+const binaryBucketName: string | undefined = process.env.BINARY_BUCKET_NAME;
+const outputFile: ExportOutput = JSON.parse(readFileSync('migrationExport_Output.txt').toString());
+
+const s3Client: S3 = new S3({
+  region: API_AWS_REGION
+});
+
+if (!BINARY_BUCKET_NAME) {
+  throw new Error('BINARY_BUCKET_NAME environment variable not specified');
+}
+if (!EXPORT_BUCKET_NAME) {
+  throw new Error('EXPORT_BUCKET_NAME environment variable not specified');
+}
+
+// Step 1, Retrieve all Binary Files from Export Output
+function getAllBinaryKeysInFolder(folderName: string): string[] {
+  // eslint-disable-next-line security/detect-object-injection
+  return outputFile.itemNames[folderName].filter((x) => x.includes('Binary'));
+}
+// Step 2, download files from S3 to get Ids
+async function getBinaryResource(itemKey: string): Promise<GetObjectOutput> {
+  console.log(`getting ${itemKey}`);
+  const file = await s3Client
+    .getObject({
+      Bucket: EXPORT_BUCKET_NAME!,
+      Key: itemKey
+    })
+    .promise();
+  if (file.$response.error) {
+    throw new Error(`Failed to get object ${itemKey}`);
+  }
+  return file;
+}
+
+// Step 3, Retrieve Binary objects from S3 Binary Bucket
+async function getBinaryObject(itemKey: string, versionId: number = 1): Promise<GetObjectOutput> {
+  const files = await s3Client
+    .listObjectsV2({
+      Bucket: BINARY_BUCKET_NAME!,
+      Prefix: `${itemKey}_${versionId}`
+    })
+    .promise();
+  if (files.$response.error || !files.Contents || files.Contents.length === 0) {
+    throw new Error(`Failed to find object ${itemKey}`);
+  }
+  // there should only be 1 file in the S3 bucket that begins with id_vid.
```

Comment:
Can we confirm that there is 1 file first?
In the default case, could there also be a bucket where the vid number starts with 1, like 10, 11, etc that would also be returned from listObjects?

Reply:
That's a really good point. I can update it to be `id_vid.` instead of `id_vid`, so that we are sure its the end of file. Does that work?

Reply:
Yep!